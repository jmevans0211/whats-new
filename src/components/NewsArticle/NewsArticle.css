article {
  border: black solid 3px;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  max-width: 380px;
  padding: 20px 10px;
}

img {
  border: black double 3px;
  max-height: 200px;
}