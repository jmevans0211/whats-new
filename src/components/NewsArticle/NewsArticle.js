import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
  const { headline } = props;
  const { img } = props;
  const { description } = props;
  const { url } = props

  return (
    <article>
      <img src={ img } alt={ description }/>
      <h1>{ headline }</h1>
      <p>{ description }</p>
      <footer><a href={ url }>Link To article</a></footer>
    </article>
  )
}

export default NewsArticle;