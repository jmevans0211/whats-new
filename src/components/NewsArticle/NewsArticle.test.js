
import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      headline={}
      img={}
      description={}
      url={}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});