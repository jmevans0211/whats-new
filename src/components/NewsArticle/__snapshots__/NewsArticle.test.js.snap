// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<article>
  <div>
    <img
      alt="The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios."
      src="https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp"
    />
  </div>
  <h2>
    Spider-Man Will Remain in the Marvel Cinematic Universe
  </h2>
  <p>
    The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.
  </p>
  <footer>
    <a
      href="https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html"
    >
      Link To article
    </a>
  </footer>
</article>
`;
